web/src/components/users/FirstUser.jsx

```diff
@@ -220,11 +223,38 @@ export default function FirstUser() {
   const submitDisable = formValues.userName === "" || (isSettingPassword && !usingValidPassword);
 
   const displaySuggestions = !formValues.userName && formValues.fullName && showSuggestions;
+  useEffect(() => {
+    if (displaySuggestions) {
+      setFocusedIndex(-1);
+      setSuggestions(suggestUsernames(formValues.fullName));
+    }
+  }, [displaySuggestions, formValues.fullName]);
+
   const onSuggestionSelected = (suggestion) => {
     setInsideDropDown(false);
     setFormValues({ ...formValues, userName: suggestion });
   };
 
+  const handleKeyDown = (event) => {
+    switch (event.key) {
+      case 'ArrowDown':
+        event.preventDefault(); // Prevent page scrolling
+        setFocusedIndex((prevIndex) => (prevIndex + 1) % suggestions.length);
+        break;
+      case 'ArrowUp':
+        event.preventDefault(); // Prevent page scrolling
+        setFocusedIndex((prevIndex) => (prevIndex - 1 + suggestions.length) % suggestions.length);
```

Comment:
With this code, when pressing the <kbd>Arrow Up</kbd> when suggestion does not have the focus yet, the next-to-last suggestion get focused instead of the last one.

For example, having 4 suggestions:

```
FirstUser.jsx:247 prevIndex: -1
FirstUser.jsx:248 suggestions.length: 4
FirstUser.jsx:249 result of (prevIndex - 1 + suggestions.length) % suggestions.length: 2
```